java/memory/memory-core/src/main/java/org/apache/arrow/memory/ArrowBuf.java

```diff
@@ -1033,7 +1034,7 @@ public void getBytes(long index, OutputStream out, int length) throws IOExceptio
       // copy length bytes of data from this ArrowBuf starting at
       // address addr(index) into the tmp byte array starting at index 0
       byte[] tmp = new byte[length];
-      MemoryUtil.UNSAFE.copyMemory(null, addr(index), tmp, MemoryUtil.BYTE_ARRAY_BASE_OFFSET, length);
```

Comment:
nit: is it possible to make the `UNSAFE` constant `private` to avoid other problems like this? (We can do that in a separate ticket.)